src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -506,9 +506,19 @@ void StorageReplicatedMergeTree::waitMutationToFinishOnReplicas(
     if (replicas.empty())
         return;
 
+    /// Current replica must always be present in the list as the first element because we use local mutation status
```

Comment:
@davenger maybe better to throw if it is not there? That said that I don't see how it is possible right now.